wm/client.go

```diff
@@ -65,6 +65,15 @@ func (c *client) IconName() string {
 	return windowIconName(c.wm.x, c.win)
 }
 
+func (c *client) Icon() fyne.Resource {
+	var icon fyne.Resource = nil
```

Comment:
This might be your problem... try moving the return into the if statement - the icon var is not needed...